lang/rust/rust.talon

```diff
@@ -0,0 +1,69 @@
+mode: command
+and mode: user.rust
+mode: command
+and mode: user.auto_lang
+and code.language: rust
+-
+tag(): user.code_imperative
+tag(): user.code_object_oriented
+
+tag(): user.code_comment_line
+tag(): user.code_comment_documentation
+tag(): user.code_data_bool
+tag(): user.code_data_null
+tag(): user.code_functions
+tag(): user.code_functions_gui
+tag(): user.code_libraries
+tag(): user.code_libraries_gui
+tag(): user.code_operators_array
+tag(): user.code_operators_assignment
+tag(): user.code_operators_bitwise
+tag(): user.code_operators_math
+
+settings():
+    user.code_private_function_formatter = "SNAKE_CASE"
+    user.code_protected_function_formatter = "SNAKE_CASE"
+    user.code_public_function_formatter = "SNAKE_CASE"
+    user.code_private_variable_formatter = "SNAKE_CASE"
+    user.code_public_variable_formatter = "SNAKE_CASE"
+
+^function <user.text>$: user.code_default_function(text)
+^public function <user.text>$: user.code_default_function(text)
+
+unsafe: user.code_state_unsafe()
+
+^struct <user.text>:
+    insert("struct ")
+    insert(user.formatted_text(text, "hammer"))
+    insert(" {\n\n}")
+    edit.left()
+    edit.up()
+    key(tab)
+
+^enum <user.text>:
+    insert("enum ")
+    insert(user.formatted_text(text, "hammer"))
+    insert(" {\n\n}")
+    edit.left()
+    edit.up()
+    key(tab)
+
+toggle use: user.code_toggle_libraries()
+
+state let:
```

Comment:
I think for knausj just use `state let: "let"`. It looks like other languages with variable keywords (javascript, go) only add that keyword. It'd be good to be consistent.